wallet/db.c

```diff
@@ -422,16 +420,22 @@ void db_commit_transaction(struct db *db)
 /**
  * db_open - Open or create a sqlite3 database
  */
-static struct db *db_open(const tal_t *ctx, char *filename)
+static struct db *db_open(const tal_t *ctx, const char *network_name)
 {
 	int err;
 	struct db *db;
 	sqlite3 *sql;
+	const char *filename;
 
 	if (SQLITE_VERSION_NUMBER != sqlite3_libversion_number())
 		fatal("SQLITE version mismatch: compiled %u, now %u",
 		      SQLITE_VERSION_NUMBER, sqlite3_libversion_number());
 
+	filename = tal_fmt(tmpctx, "%s.sqlite3", network_name);
+#ifdef COMPAT_V052
+	/* Earlier versions didn't include network name */
+	rename("lightningd.sqlite3", filename);
```

Comment:
Similar to previous comment regarding `hsm_secret`. We should take care to inform users before merging this PR as much as possible (maybe send an email on the mailing list? It is rare enough that hopefully people will notice?)